test/e2e/v1beta1/scripts/gh-actions/run-e2e-experiment.py

```diff
@@ -0,0 +1,272 @@
+        config.load_kube_config()
+        for _ in range(10):
+
+            services = client.CoreV1Api().list_namespaced_service(exp_namespace)
+            for i in services.items:
+                is_deleted = 0 if i.metadata.name == resource_name else 1
+
+            deployments = client.AppsV1Api().list_namespaced_deployment(exp_namespace)
+            for i in deployments.items:
+                is_deleted *= 0 if i.metadata.name == resource_name else 1
+            if is_deleted == 1:
+                break
+
+            # Deployment and Service deletion might take some time.
+            time.sleep(1)
+
+        if is_deleted == 0:
+            raise Exception(
+                "Suggestion Service or Deployment is still alive for "
+                f"ResumePolicy == {experiment.spec.resume_policy}. "
+                f"Alive Services: {[i.metadata.name for i in services.items]}. "
+                f"Alive Deployments: {[i.metadata.name for i in deployments.items]}."
+            )
+
+        # For FromVolume resume policy PVC should not be deleted.
+        if experiment.spec.resume_policy == "FromVolume":
+            PVCs = client.CoreV1Api().list_namespaced_persistent_volume_claim(
+                exp_namespace
+            )
+            is_deleted = 1
+            for i in PVCs.items:
+                if i.metadata.name == resource_name:
+                    is_deleted = 0
+            if is_deleted == 1:
+                raise Exception(
+                    "PVC is deleted for FromVolume resume policy. "
+                    f"Alive PVCs: {[i.metadata.name for i in PVCs.items]}."
+                )
+
+
+def run_e2e_experiment(
+    katib_client: KatibClient,
+    experiment: models.V1beta1Experiment,
+    exp_name: str,
+    exp_namespace: str,
+):
+
+    # Create Katib Experiment and wait until it is finished.
+    print(
+        "\n\n\nCreating Experiment: {}/{} with MaxTrialCount: {}, ParallelTrialCount: {}".format(
+            exp_namespace,
+            exp_name,
+            experiment.spec.max_trial_count,
+            experiment.spec.parallel_trial_count,
+        )
+    )
+
+    # Wait until Experiment reaches Succeeded condition.
+    katib_client.create_experiment(experiment, exp_namespace)
+    experiment = katib_client.wait_for_experiment_condition(
+        exp_name, exp_namespace, timeout=EXPERIMENT_TIMEOUT
+    )
+
+    # Test resume feature for "FromVolume" and "LongRunning" Experiments.
+    # TODO (andreyvelich): Once we change the default resume policy to "Never",
+    # also test restart for "LongRunning" Experiment here instead of "random".
+    # Ref: https://github.com/kubeflow/katib/issues/2055
+    if exp_name == "from-volume-resume" or exp_name == "random":
+        max_trial_count = experiment.spec.max_trial_count + 1
+        parallel_trial_count = experiment.spec.parallel_trial_count + 1
+        print(
+            f"Restarting Experiment {exp_namespace}/{exp_name} "
```

Comment:
Why is this `Restarting experiment` for `random`?

Reply:
@johnugeorge We also resume Experiment for [random search Experiment to test `LongRunning`](https://github.com/kubeflow/katib/blob/master/test/e2e/v1beta1/hack/gh-actions/run-e2e-experiment.go#L102) Experiment.